mtt/src/mtt.rs

```diff
@@ -109,6 +110,52 @@ mod mtt_const {
     pub const MTT_INDEX_SHIFT: u64 = 26;
 }
 
+// This is to help facilitate testing. The issue is that when testing, we use the PFN
+// of the VA for the L1-page, and since it tends to be very large (unlike the actual PA),
+// it might sets bits 35:34 when converting to a L1 page (thereby converting the L2 entry
+// into a different type). We avoid the issue by shifting everything left by 4.
```

Comment:
ugh, this is unfortunate.

can we do a better job defining the constants in terms of each other, so we don't have to repeat the entire set of definitions?